pkg/recorder/recorder.go

```diff
@@ -85,12 +88,40 @@ func (r *Recorder) Record(rec record.Record) error {
 	}
 	// we want to record our metadata file anyway
 	if r.size+recordSize > r.maxArchiveSize && rec.Name != MetadataRecordName {
-		return fmt.Errorf("record %s(size=%d) exceeds the archive size limit %d and will not be included in the archive",
-			recordName, recordSize, r.maxArchiveSize)
+		errs = append(errs, fmt.Errorf(
+			"record %s(size=%d) exceeds the archive size limit %d and will not be included in the archive",
+			recordName, recordSize, r.maxArchiveSize,
+		))
+		return errs
 	}
-	r.records[memoryRecord.Name] = memoryRecord
+
+	if existingRecord, found := r.records[memoryRecord.Name]; found {
+		errs = append(errs, fmt.Errorf(
+			`the record %v with the same name was already recorded, overwriting with the record %v`,
+			existingRecord.Print(), memoryRecord.Print(),
+		))
+		r.size -= int64(len(existingRecord.Data))
+	}
+
 	r.size += recordSize
-	return nil
+	r.records[memoryRecord.Name] = memoryRecord
+
+	if existingPath, found := r.recordedFingerprints[fingerprint]; found {
+		existingRecord, found := r.records[existingPath]
+		if !found {
+			existingRecord = &record.MemoryRecord{Name: "unable to find the record"}
+		}
+		// this doesn't necessarily mean it's an error. There can be a collision after hashing
+		errs = append(errs, &types.Warning{UnderlyingValue: fmt.Errorf(
+			`the record %v with the same fingerprint %v was already recorded, `+
+				`recording another one with a different path %v`,
```

Comment:
It's just a nit, but I got message like:
> "warning: the record MemoryRecord{Name: \"config/configmaps/openshift-config/admin-kubeconfig-client-ca/ca-bundle.crt\", At: \"2022-04-28 14:38:29.852686256 +0200 CEST m=+54.467188189\", len(Data): 65, Fingerprint: \"34cdf5d36d45a9e3ee7c2be31546e3d937ae4ebf144ac07b0210c06a1610a94d\"} with the same fingerprint 34cdf5d36d45a9e3ee7c2be31546e3d937ae4ebf144ac07b0210c06a1610a94d was already recorded, recording another one with a different path MemoryRecord{Name: \"config/configmaps/openshift-config/etcd-ca-bundle/ca-bundle.crt\", At: \"2022-04-28 14:38:29.852757424 +0200 CEST m=+54.467259358\", len(Data): 65, Fingerprint: \"34cdf5d36d45a9e3ee7c2be31546e3d937ae4ebf144ac07b0210c06a1610a94d\"}",

which is pretty verbose IMO. The same fingerprint is mentioned three times. I'm not sure the time and length of the data is of any use.